marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java

```diff
@@ -343,7 +357,97 @@ public void upgradeProject() throws IOException {
     @Override public String getUserModulesDeployTimestampFile() {
         return Paths.get(projectDirString, ".tmp", USER_MODULES_DEPLOY_TIMESTAMPS_PROPERTIES).toString();
     }
-
+    
+    //copying only the required old hub-internal-config db and server files to new locations
+    //the security files (users, roles, privileges etc are not copied from old hub-internal-config)
+    private void upgradeHubInternalConfig(Path sourceDir, Path renamedSourceDir, Set<String> hubConfigFiles) throws IOException {
+        if (Files.exists(sourceDir)) {
+            Files.walk(Paths.get(sourceDir.toUri()))
+            .filter( f -> !Files.isDirectory(f))
+            .forEach(f -> {
+                String path = f.toFile().getName();
+                //copy jobs,staging db to hub-internal-config
+                if(path.toLowerCase().equals("job-database.json")
+                || path.toLowerCase().equals("staging-database.json")) {
+                    try {
+                        FileUtils.copyFile(f.toFile(), getHubDatabaseDir().resolve(f.getFileName()).toFile());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+              //copy modules,final db to ml-config
+                if(path.toLowerCase().equals("modules-database.json")
+                || path.toLowerCase().equals("final-database.json")) {
+                    try {
+                        FileUtils.copyFile(f.toFile(), getUserDatabaseDir().resolve(f.getFileName()).toFile());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+              //copy jobs,staging server to hub-internal-config
+                if(path.toLowerCase().equals("job-server.json")
+                || path.toLowerCase().equals("staging-server.json")) {
+                    try {
+                        FileUtils.copyFile(f.toFile(), getHubServersDir().resolve(f.getFileName()).toFile());
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+              //copy final server to ml-config
+                if(path.toLowerCase().equals("final-server.json")) {
+                    try {
+                        FileUtils.copyFile(f.toFile(), getUserServersDir().resolve(f.getFileName()).toFile());
```

Comment:
Same thing here, what if this file already exists?

I'll do some testing with a 3.0 build. I think the only reason the files would exist under src/main/ml-config already is if someone first ran hubInit using 4.1, then hubUpgrade. But it would be fine to overwrite the files, because the user wouldn't lose anything.